app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -71,17 +72,51 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
             return self.dblclick(d);});
         added.append("image")
             .attr("xlink:href",function(d) {
-                return "/assets/svg/" + self.class_name(d) + ".svg";
+                var iconInfo = self.getIcon(d);
+                switch(iconInfo.type) {
+                    case 'image':
+                        return iconInfo.icon;
+                    case "glyph": // the icon will be provided as a text
+                        return null;
+                }
             })
-            .attr("y", function(d) { return self.getDimensions(d).y})
-            .attr("x", function(d) { return self.getDimensions(d).x})
-            .attr("height", function(d) { return self.getDimensions(d).height})
-            .attr("width", function(d) { return self.getDimensions(d).width});
+            .attr("y", function(d) { return self.getCircleDimensions(d).y})
+            .attr("x", function(d) { return self.getCircleDimensions(d).x})
+            .attr("height", function(d) { return self.getCircleDimensions(d).height})
+            .attr("width", function(d) { return self.getCircleDimensions(d).width});
+        // attached labels
         added.append("text")
             .attr("x", 26)
             .attr("y", 24)
-            .text(function(d) { return d.item.name }).style("font-size", function(d) {return "12px"}).style("fill", function(d) {return "black"})
-            .style("display", function(d) {if ($scope.checkboxModel.value) {return "block"} else {return "none"}});
+            .text(function(d) {
+                return d.item.name;
+            })
+            .attr('class', 'attached-label')
+            .style("font-size", function(d) {
```

Comment:
is this needed given the following css?
https://github.com/ManageIQ/manageiq/blob/master/app/assets/stylesheets/topology.css#L47